### PR TITLE
Updating github actions and adding prettier

### PR DESCRIPTION
**Pull request by CLeopard99**
## Description
https://scottlogic.atlassian.net/browse/FS-52
Github actions were already set up, have updated to use a maintained version of ruff, run actions on merge/push, and added prettier for the frontend

## Changelog
- ruff-action updated as [chartboost/ruff-action@v1](https://github.com/ChartBoost/ruff-action) has been archived
- pushes and merges will now run actions so we can check branches are working
- adding prettier to format the frontend in actions, replacing the eslint rules and has done an initial format

### CODE REVIEW

File: .github/workflows/lint-backend.yml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 name: Lint Backend
-on: 
-  pull_request:
+on: [push, pull_request]
```
Comment (mic-smith) [inline .github/workflows/lint-backend.yml:2]: The push will trigger for pushes to all branches. e.g. this branch ran the checks twice, once for the push and the once pr.
Do we want to target the push just to `main` (and add any other protected branches later once branch naming confirmed) ?

File: .github/workflows/lint-frontend.yml [hunk 2]
```diff
@@ -23,4 +21,6 @@ jobs:
         run: npm install
 
       - name: Lint
-        run: npm run lint
\ No newline at end of file
+        run: |
+          npm run lint
+          npm run format
```
Comment (mic-smith) [inline .github/workflows/lint-frontend.yml:26]: `npm run format` will reformat the files.
I think we just want to check here instead? e.g. `npx prettier . --check`  or an equivalent script in package.json

File: frontend/.prettierrc.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "singleQuote": true,
+  "trailingComma": "es5",
+  "tabWidth": 2,
+  "semi": true,
```
Comment (mic-smith) [inline frontend/.prettierrc.json:5]: The `"tabWidth": 2` and  `"semi": true,` are the defaults, so could remove these.

File: .github/workflows/lint-frontend.yml (hunk 2 above)
Comment (CLeopard99) [reply .github/workflows/lint-frontend.yml:26]: The reformat was intended, if we make it just a check, then we will probably need to add format in a pre-push or pre-commit. Is there any benefit in not just doing it actions?

File: .github/workflows/lint-backend.yml (hunk 1 above)
Comment (CLeopard99) [reply .github/workflows/lint-backend.yml:2]: Yeah wasn't sure about this, probably some benefits to having it but maybe not worth the unsightly 8 checks in PRs.

File: .github/workflows/lint-backend.yml [hunk 4]
```diff
@@ -14,6 +13,6 @@ jobs:
         uses: actions/setup-python@v2
 
       - name: Lint
-        uses: chartboost/ruff-action@v1
+        uses: astral-sh/ruff-action@v1
```
Comment (IMladjenovic) [inline .github/workflows/lint-backend.yml:16]: I meant to ask earlier. What is `ruff-action`?

File: .husky/pre-commit [hunk 5]
```diff
@@ -0,0 +1,3 @@
+cd frontend
+npx lint-staged
+git add .
```
Comment (mic-smith) [inline .husky/pre-commit:3]: This would add any untracked changes too. 
I think the pre-commit hook should just check, rather than update and add files?

Comment (CLeopard99) [reply .husky/pre-commit]: Changed it to just check so users will have to format themselves

**Review by mic-smith — APPROVED** [review]

File: .husky/pre-commit [hunk 6]
```diff
@@ -0,0 +1,2 @@
+cd frontend
+npx lint-staged
```
Comment (IMladjenovic) [inline .husky/pre-commit:2]: Open to this but I think it's worth discussing with the team whether we want linting running on pre-commit hooks.

Comment (IMladjenovic) [reply .husky/pre-commit:2]: Actually, since it's frontend only, this can be a decision for just me, you and Mike. What are your thoughts @mic-smith?

Comment (mic-smith) [reply .husky/pre-commit:2]: Maybe having the linting as a pre-push hook might be better?

Comment (CLeopard99) [reply .husky/pre-commit:2]: Can do, I have no strong feelings towards either

File: .husky/pre-push [hunk 7]
```diff
@@ -0,0 +1,4 @@
+cd frontend
+npx lint-staged
+cd ../backend
+ruff check
```
Comment (IMladjenovic) [inline .husky/pre-push:4]: This has a wider impact on the backend team, I would add this to the retro board to discuss with the team as a whole. Happy for frontend pre-push/pre-commit hooks to be agreed between you, me and Mike

Comment (IMladjenovic) [reply .husky/pre-push:4]: So, for now, I would remove line 3-4 and keep 1-2 :)

**Review by IMladjenovic — APPROVED** [review]

**Pull request merged**